src/imageLoader/loadImageSets.js

```diff
@@ -31,34 +31,67 @@ async function getImageSets(datastoreId, config, awsFilter, _nextToken = '') {
     return imageSetsMetadataSummaries;
 }
 
+const loadImageSetsCache = new Map();
 const loadImageSets = async (config, filters) => {
+    const startTime = performance.now();
     const aswFilter = Object.keys(filters).reduce((obj,key) =>{
         const awsKey = ohifToAwsParam[key];
         if(awsKey) {
             obj[awsKey] = filters[key];
         }
         return obj;
     },{});
+    const cacheKey = JSON.stringify(aswFilter);
+    // Todo make it configurable
+    setTimeout(() => {
+        loadImageSetsCache.delete(cacheKey);
+    }, 1000 * 60 * 5);
+    const cached = loadImageSetsCache.get(cacheKey);
+    if (cached) {
+        performance.measure(
+            "healthlake:imagessets-load-from-cache", {
+                start: startTime,
+                detail: "cached",
+                end: performance.now(),
+            }
+        );
+        return cached;
+    } else {
+        const getImageSetPromise = getImageSets(config.datastoreID, config, aswFilter).then((imageSetsMetadataSummaries) => {
+            performance.measure(
+                "healthlake:imagessets-load", {
+                    start: startTime,
+                    detail: "request",
+                    end: performance.now(),
+                }
+            );
+            const json = imageSetsMetadataSummaries.map(map.bind(null, config.datastoreID));
+            const uniq = reduceImageSetsByStudy(json);
+            config.collections[json.ImageSetID] = uniq;
+            return uniq;
+        });
+        loadImageSetsCache.set(cacheKey, getImageSetPromise);
+        return getImageSetPromise;
+    }
+}
 
-    const startTime = performance.now();
-    const imageSetsMetadataSummaries = await getImageSets(config.datastoreID, config, aswFilter);
-    performance.measure(
-        "healthlake-imagessets-load", {
-            start: startTime,
-            end: performance.now(),
+function reduceImageSetsByStudy(json) {
+    return Object.values(json.reduce((cc, a) => {
+        if (!cc[a['0020000D'].Value[0]]) {
```

Comment:
Mapping it by studying Uid and grouping by ImagesetId here is the fix for the multiple series